### PR TITLE
Appliance and ad support

### PR DESCRIPTION
**Pull request by Overglazed**
Add Appliance and Active Directory data sources to the ecloud terraform provider.

This functionality enables the ability to define Appliances and Domains by name, rather than ID.

### CODE REVIEW

File: ecloud/data_source_active_directory.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+func dataSourceActiveDirectory() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceActiveDirectoryRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"ad_domain_id": &schema.Schema{
```
Comment (0x4c6565) [inline ecloud/data_source_active_directory.go]: We don't need to reference `ad_domain_id` here, as this is a duplicate of the datasource ID

File: ecloud/data_source_appliance.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+func dataSourceAppliance() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceApplianceRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"appliance_id": &schema.Schema{
```
Comment (0x4c6565) [inline ecloud/data_source_appliance.go]: We don't need to reference `appliance_id` here, as this is a duplicate of the datasource ID

File: ecloud/data_source_pod_appliance.go [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"pod_id": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"appliance_id": &schema.Schema{
```
Comment (0x4c6565) [inline ecloud/data_source_pod_appliance.go]: We don't need to reference `appliance_id` here, as this is a duplicate of the datasource ID

File: ecloud/data_source_active_directory.go (hunk 1 above)
Comment (Overglazed) [reply ecloud/data_source_active_directory.go]: removed :)

File: ecloud/data_source_appliance.go (hunk 2 above)
Comment (Overglazed) [reply ecloud/data_source_appliance.go]: removed

File: ecloud/data_source_pod_appliance.go (hunk 3 above)
Comment (Overglazed) [reply ecloud/data_source_pod_appliance.go]: removed

**Comment by 0x4c6565** [comment]
LGTM, thank you!

**Pull request merged**